src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023holostiminterface.py

```diff
@@ -0,0 +1,242 @@
+from collections import defaultdict
+from copy import deepcopy
+from pathlib import Path
+from typing import Optional
+
+import numpy as np
+import h5py
+from neuroconv import BaseDataInterface
+from neuroconv.tools.roiextractors.roiextractors import get_default_segmentation_metadata
+from neuroconv.utils import FolderPathType, FilePathType, get_base_schema, get_schema_from_hdmf_class
+from pynwb import NWBFile
+from pynwb.device import Device
+from pynwb.ogen import OptogeneticStimulusSite
+from pynwb.ophys import PlaneSegmentation, OpticalChannel
+from natsort import natsorted
+from roiextractors.extractors.tiffimagingextractors.scanimagetiff_utils import (
+    parse_metadata,
+    extract_extra_metadata,
+    extract_timestamps_from_file,
+)
+from ndx_patterned_ogen import (
+    PatternedOptogeneticStimulusTable,
+    OptogeneticStimulusTarget,
+    TemporalFocusing,
+    PatternedOptogeneticStimulusSite,
+    SpatialLightModulator2D,
+    LightSource,
+)
+
+
+class Abdeladim2023HolographicStimulationInterface(BaseDataInterface):
+    """
+    Data Interface for writing holographic photostimulation data for the MouseV1 to NWB file using Abdeladim2023HolographicStimulationInterface.
+    """
+
+    def __init__(
+        self,
+        folder_path: FolderPathType,
+        holographic_stimulation_file_path: FilePathType,
+        epoch_name: str = None,
+        targeted_plane_segmentation_name: Optional[str] = None,
+        verbose: bool = True,
+    ):
+        """
+        Data interface for adding the holographic stimulation to the NWB file.
+
+        Parameters
+        ----------
+        folder_path : FolderPathType
+            The folder path that contains the holographic stimulation data.
```

Comment:
[minor] I think it would make sense to clarify this the folder with the imaging data as well, otherwise it is confusing how different it is from `holographic_stimulation_file_path` which does not have a docstring.

Reply:
Good catch, thanks! it's updated now